### PR TITLE
Add install instructions

### PR DESCRIPTION
**Pull request by shwina**
@wking @gvwilson @tbekolay 

Here's a first draft of `install.md`.

### CODE REVIEW

**Comment by wking** [comment]
On Mon, Feb 29, 2016 at 06:05:59PM -0800, Ashwin Srinath wrote:
>     M index.md (1)
>     A install.md (25)

You'll also want to bump LAYOUT.md to remove the “Software
installation and configuration instructions *aren't* in the lesson…”
note.  And the “Pages” section will need an install.md entry (just
after index.md?).

It looks like Pandoc adds header IDs automatically, so this is
probably all we need.  A note about ID stability might be in order,
since we'll be encouraging lessons to link other lesson's installation
instructions.

It might be worth filling this in with a more detailed example
(e.g. Pandoc installation instructions, to pick a lesson-example
dependency ;) to give folks some guidance for local depth vs. linking
to external docs.  Of course, lesson maintainers would be free to pick
their own level of depth, but some guidelines and an example here
would make it more likely that we end up with a uniform feel.

**Comment by shwina** [comment]
@wking - does this look better?

**Comment by wking** [comment]
On Tue, Mar 01, 2016 at 08:56:52AM -0800, Ashwin Srinath wrote:
> @wking - does this look better?

50ac2f02 looks good to me.  @cmacdonell?

**Comment by cmacdonell** [comment]
Yup, looks fine

**Comment by tbekolay** [comment]
I'm still in favour of a separate installation lesson, perhaps one lesson per OS. I worry that having an install.md in each repository means that we will duplicate effort as there will be multiple lessons per tool, and therefore multiple copies of the same installation instructions.

Should I make a quick repository of what that might look like in order to compare the two approaches?

**Comment by shwina** [comment]
@tbekolay 

+1, It would be helpful to see how both approaches look

File: index.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Paragraph of introductory material.
 
 ## Other Resources
 
+*   [Install instructions](install.html)
```
Comment (gvwilson) [inline index.md]: "Installation" rather than "Install"

File: install.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+layout: page
+title: Lesson Title
+subtitle: Install instructions
```
Comment (gvwilson) [inline install.md]: "Installation" rather than "Install"

**Comment by gvwilson** [comment]
I prefer the per-OS layout.

**Comment by cmacdonell** [comment]
@tbekolay We have a small working group ( @shwina, @wking and myself) that are spearheading the development of install instructions.  Based on the "Installation instructions" discussion a few days ago on the Discuss list:

http://lists.software-carpentry.org/pipermail/discuss_lists.software-carpentry.org/2016-February/

We have decided to start with per-lesson install.md.  We will then build a stand-alone "Installation Lesson" based on pulling together the per-lesson install.md.  The risk in developing both per-lesson and a stand-alone lesson concurrently is that we split our already small working group and will end up with two incomplete approaches.

**Comment by tbekolay** [comment]
My understanding from the mailing list was that the working group would come up with a proposal to present to the maintainers list to vote on? I don't think anyone expects the working group to do all of the work...

**Comment by gvwilson** [comment]
Yes - please come up with a proposal for the maintainers to vote on, 
then we can share the load.

**Comment by cmacdonell** [comment]
Ok.  Can someone create an install-lesson repo under the swcarpentry account to build from?

**Comment by cmacdonell** [comment]
We've got one - https://github.com/swcarpentry/install-lesson , thanks @gvwilson

**Comment by gvwilson** [comment]
We live to serve.

**Comment by wking** [comment]
On Wed, Mar 02, 2016 at 04:30:42AM -0800, Trevor Bekolay wrote:
> I worry that having an install.md in each repository means that we
> will duplicate effort as there will be multiple lessons per tool,
> and therefore multiple copies of the same installation instructions.

I think this is easy to avoid by linking between lessons [1], just
like 50ac2f02 is linking out to Pandoc's instructions.  I'm happy to
add a note about link-anchor stability [2], but beyond that I think
this per-lesson proposal won't require any duplication.

[1]: http://lists.software-carpentry.org/pipermail/maintainers_lists.software-carpentry.org/2016-February/000173.html
     Subject: Re: [Maintainers] setup instructions
     Date: Wed, 24 Feb 2016 11:55:05 -0800
     Message-ID: <20160224195505.GT4265@odin.tremily.us>
[2]: https://github.com/swcarpentry/lesson-example/pull/24#issuecomment-190563127

**Comment by wking** [comment]
Reroll now that we have a consensus that includes this as one option
[1]?

[1]: http://lists.software-carpentry.org/pipermail/maintainers_lists.software-carpentry.org/2016-March/000217.html
     Subject: Re: [Maintainers] Setup instructions [Final_2.rev(5).comments]
     Date: Thu, 17 Mar 2016 14:50:36 -0400
     Message-ID: <CANLckLvfRv4RZ3dHfhcEP7NH_QfR91noNaZoP3sYsFVsh6orAw@mail.gmail.com>

**Comment by wking** [comment]
On Thu, Mar 17, 2016 at 01:04:54PM -0700, Ashwin Srinath wrote:
> Closed #24.

Closed instead of rerolled?  Where are we documenting the new
workflow?

**Comment by shwina** [comment]
Sorry! Any suggestions for where to do that?

**Comment by wking** [comment]
On Sat, Mar 19, 2016 at 03:54:36PM -0700, Ashwin Srinath wrote:
> Sorry! Any suggestions for where to do that?

Non-core lessons will still use the install.md stuff you've setup
here.  Probably add a note in LAYOUT.md and “Getting ready” about
linking and how the rules are different for core and non-core lessons.

**Comment by shwina** [comment]
@wking - how does this look?

File: index.md [hunk 3]
```diff
@@ -25,3 +30,5 @@ Paragraph of introductory material.
 *   [Reference](reference.html)
 *   [Discussion](discussion.html)
 *   [Instructor's Guide](instructors.html)
+
+[swc-workshop-template-rendered]: http://swcarpentry.github.io/workshop-template/index.html
```
Comment (wking) [inline index.md]: I'd prefer https://swcarpentry.github.io/workshop-template/ for more security and a shorter URL.  Although it looks like we need to use HTTPS or a local image for the SWC banner in the workshop-template to make that work cleanly.

File: LAYOUT.md [hunk 4]
```diff
@@ -99,8 +99,12 @@ The root directory holds the content of the lesson, and must contain:
     this lesson, and is about the filesystem. (Note that we use hyphens
     rather than underscores in filenames.) See "Topics" below.
 
-4.  `install.md`: instructions for installing the software tools used
-    by learners for this lesson.
+4.  `install.md`: instructions for installing the software tools
+    used by learners for this lesson.
+    Setup instructions for tools used by the core lessons are
+    already available at the
+    [workshop template][swc-workshop-template-rendered] page.
```
Comment (wking) [inline LAYOUT.md]: “… are already available…” is pretty optimistic (and installation instructions change with time ;).  How about focusing on the rationale here:

>  Setup instructions for tools used by the core lessons should be [posted to the workshop-template repository][1], which provides a central location for documenting common tools.  Setup instructions for other lessons should be provided in `install.md`, which scales better as the number of non-core lessons increases.

[1]: https://github.com/swcarpentry/workshop-template/pulls
[2]: https://swcarpentry.github.io/workshop-template/

File: index.md [hunk 5]
```diff
@@ -13,6 +13,11 @@ Paragraph of introductory material.
 >
 > A short paragraph describing
 > what learners need to do before tackling this lesson.
+> Importantly, you must link to the
+> setup instructions provided in `install.md`
+> and/or on the [workshop template][swc-workshop-template-rendered]
+> page.
```
Comment (wking) [inline index.md]: Maybe unpack this into more explicit detail?

> Importantly, you must link to installation instructions for any tools used in this lesson.
> Tools used by core lessons will be [documented in the workshop-template][1]; link there if your tool is listed.
> If your tool is not listed there, core-lesson maintainers should [file a workshop-template pull request adding it][2].
> Non-core-lesson maintainers should document the installation of non-core tools in a local `install.md`.

Or something like that.

[1]: https://swcarpentry.github.io/workshop-template/
[2]: https://github.com/swcarpentry/workshop-template/pulls

**Comment by shwina** [comment]
Thanks @wking - incorporated your suggestions.

File: LAYOUT.md [hunk 6]
```diff
@@ -441,3 +445,4 @@ structure is:
 
 [swc-lesson-example]: https://github.com/swcarpentry/lesson-example
 [swc-lesson-template]: https://github.com/swcarpentry/lesson-template
+[swc-workshop-template-pulls]: https://github.com/swcarpentry/workshop-template/pulls
```
Comment (wking) [inline LAYOUT.md]: nit: There's some trailing whitespace here.  Found with:

    $ git diff --check origin/gh-pages..origin/pr/24
    LAYOUT.md:448: trailing whitespace.
    +[swc-workshop-template-pulls]: https://github.com/swcarpentry/workshop-template/pulls

File: index.md [hunk 7]
```diff
@@ -21,6 +33,10 @@ Paragraph of introductory material.
 
 ## Other Resources
 
+*   [Installation instructions](install.html)
 *   [Reference](reference.html)
 *   [Discussion](discussion.html)
 *   [Instructor's Guide](instructors.html)
+
+[swc-workshop-template]: http://swcarpentry.github.io/workshop-template/index.html
```
Comment (wking) [inline index.md]: This still needs an HTTPS link (suggesting students follow installation instructions retrieved over HTTP is not secure).

**Comment by wking** [comment]
I hadn't noticed 7dfe8c6 → 2c3e580 go in.  I can't see anything left
to polish with 2c3e580, so as far as I'm concerned this can get merged
whenever.

**Comment by gvwilson** [comment]
I think the maintainers have decided to go in a different direction.

**Comment by wking** [comment]
On Fri, Jun 17, 2016 at 04:12:20AM -0700, Greg Wilson wrote:
> I think the maintainers have decided to go in a different direction.

I thought this was the direction they decided to go in [1] ;).  Can
you link to the post-March decision?

[1]: https://github.com/swcarpentry/lesson-example/pull/24#issuecomment-198034573

**Pull request closed without merging**